java-llms/titanMLConnector/com/titanml/llm/TitanMLLLMConnector.java

```diff
@@ -31,53 +27,113 @@
 public class TitanMLLLMConnector extends CustomLLMClient {
     final private static DKULogger logger = DKULogger.getLogger("dku.llm.titanml");
     ResolvedSettings resolvedSettings;
+    private String readerID;
     private ExternalJSONAPIClient client;
 
     public TitanMLLLMConnector() {
     }
 
+    public void setHeaders(String key, String value){
+        this.client.addHeader(key, value);
+    }
+
     public void init(ResolvedSettings settings) {
+        logger.info("Initializing TitanMLLLMConnector-----------------------------------");
         // Initialize the TitanMLLLMConnector. Takes a ResolvedSettings object.
         this.resolvedSettings = settings;
         String endpointUrl = resolvedSettings.config.get("endpoint_url").getAsString();
+
+
         // Create a Dataiku ExternalJSONAPI client to call takeoff with
         Consumer<HttpClientBuilder> customizeBuilderCallback = (builder) -> {
             builder.setRedirectStrategy(new LaxRedirectStrategy());
             HTTPBasedLLMNetworkSettings networkSettings = new HTTPBasedLLMNetworkSettings();
             OnlineLLMUtils.add429RetryStrategy(builder, networkSettings);
         };
         client = new ExternalJSONAPIClient(endpointUrl, null, true, null, customizeBuilderCallback);
+
+        String consumer_group = null;
+        if (settings.config.get("consumer_group") != null) {
+            consumer_group = settings.config.get("consumer_group").getAsString();
+        }
+
+        if (consumer_group == null || consumer_group.isEmpty()) {
+            logger.info("No consumer group was specified");
+            consumer_group = "primary";
```

Comment:
[nitpick] 

I like the default `consumer_group`, this is a nice addition 👍 

I always like to make it clear in the logs when we modify the user's input.

```suggestion
            logger.info("No consumer group was specified, defaulting to 'primary'");
            consumer_group = "primary";
```